### ISSUE TITLE
brew test lab fails

### ISSUE THREAD

**Issue by MartinDelille** [zaquestion/lab#291]
I wanted to push the `lab` formula to https://github.com/homebrew/homebrew-core: https://github.com/MartinDelille/homebrew-core/blob/lab/Formula/lab.rb

Unfortunately, `brew test lab` fails because the test script block on this line: https://github.com/MartinDelille/homebrew-core/blob/lab/Formula/lab.rb#L21

Here is the log when I press Ctrl+C to stop it:

```
Testing lab
==> git init
==> git add haunted house
==> git commit -a -m Initial Commit
==> git config --local --add gitlab.host http://example.com
==> git config --local --add gitlab.user test
==> git config --local --add gitlab.token test
==> /usr/local/Cellar/lab/0.15.0/bin/lab ls-files
^CError: lab: failed
Interrupt
/usr/local/Homebrew/Library/Homebrew/formula.rb:1897:in `wait'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1897:in `block in system'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1858:in `open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1858:in `system'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/lab.rb:21:in `block in <class:Lab>'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1732:in `block (3 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/utils.rb:539:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1731:in `block (2 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:870:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1730:in `block in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1973:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:55:in `block in run'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:55:in `chdir'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:55:in `run'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1972:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1724:in `run_test'
/usr/local/Homebrew/Library/Homebrew/test.rb:28:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
```

**Comment by zaquestion** [comment]
Well, I can at least that lab doesn't use `git config` as configuration anymore. the test portion of lab's formula should surely be updated. I'd also recommend grabbing `0.15.2` (we've had a couple recent patches)

Also feel like I recall homebrew-core requiring packages are built and don't reference pre-built binaries, so you might run into that as well. Perhaps I'm wrong though, couldn't easily find the documentation on it.

**Comment by MartinDelille** [comment]
I updated to the `0.15.2` and removed the `git config` commands but the script is still blocking https://github.com/MartinDelille/homebrew-core/blob/lab/Formula/lab.rb

If you look at the formula you will see that it build from source so it should pass homebrew-core maintainer requirement.

**Comment by ianfoo** [comment]
This is addressed in zaquestion/homebrew-tap#1.

```
$ brew test ./lab.rb
Testing lab
==> git init
==> git add haunted house
==> git commit -a -m Initial Commit
==> LAB_CORE_HOST=foo LAB_CORE_USER=bar LAB_CORE_TOKEN=baz /usr/local/Cellar/lab/0.15.3/bin/lab ls-files
```

**Comment by zaquestion** [comment]
@ianfoo Thanks for the hotfix, unfortunately to make the changes permanent you'll also need to make the changes here: https://github.com/zaquestion/lab/blob/master/.goreleaser.yml#L20 Which is the source of the `lab.rb` file in homebrew-tap

**Comment by bmeneguele** [comment]
AFAICT this is now fixed (by #306).
Feel free to reopen it if needed.
